zerver/lib/actions.py

```diff
@@ -367,16 +367,26 @@ def send_message_to_signup_notification_stream(
         internal_send_stream_message(sender, signup_notifications_stream, topic_name, message)
 
 
-def notify_new_user(user_profile: UserProfile) -> None:
+def notify_new_user(user_profile: UserProfile, acting_user: Optional[UserProfile] = None) -> None:
     user_count = realm_user_count(user_profile.realm)
     sender_email = settings.NOTIFICATION_BOT
     sender = get_system_bot(sender_email, user_profile.realm_id)
 
     is_first_user = user_count == 1
     if not is_first_user:
-        message = _("{user} just signed up for Zulip. (total: {user_count})").format(
-            user=f"@_**{user_profile.full_name}|{user_profile.id}**", user_count=user_count
-        )
+        # If acting_user exists
+        if acting_user:
+            message = _(
+                "{user} accepted {referrer}'s invite to join Zulip. (total: {user_count})"
```

Comment:
"... invitation to join Zulip" (rather than "invite to join") is I think clearer English.